File: reference/typescript/src/paths.ts
import { fileURLToPath } from 'node:url';
import { dirname, resolve } from 'node:path';

// Language-neutral spec artifacts live at the repo root under spec/, so both reference
// implementations validate against the same JSON Schema and conformance vectors. Resolve
// them relative to this file (reference/typescript/src/), never the cwd.
const HERE = dirname(fileURLToPath(import.meta.url)); // reference/typescript/src
const REPO_ROOT = resolve(HERE, '..', '..', '..'); // repo root

export const SPEC_SCHEMA_DIR = resolve(REPO_ROOT, 'spec', 'schema');
export const SPEC_VECTORS_DIR = resolve(REPO_ROOT, 'spec', 'vectors');

File: reference/python/src/auditable_mcp/paths.py
"""Path resolution for shared spec artifacts.

Resolves absolute paths to the language-neutral JSON Schema and conformance vectors
located in the repository's `spec/` directory.
"""

from pathlib import Path

# reference/python/src/auditable_mcp/paths.py -> parents[4] is the repository root.
_REPO_ROOT = Path(__file__).resolve().parents[4]

SPEC_SCHEMA_DIR = _REPO_ROOT / 'spec' / 'schema'
SPEC_VECTORS_DIR = _REPO_ROOT / 'spec' / 'vectors'
